src/task.py

```diff
@@ -540,10 +567,33 @@ def is_gridcell_unique(self, df):
         return df_counts.index.is_unique
 
     def calc(self):
+        print(self.df_adhoc)
+        print(self.df_signsurvey)
+        print(self.df_cameratrap)
+
+        ss_observed = self.fc_ss.filter(ee.Filter.gt(SS_SEGMENTS_DETECTED, 0))
+        ct_observed = self.fc_ct.filter(ee.Filter.gt(CT_DAYS_DETECTED, 0))
+        ah_observed = self.fc_adhoc
+        observed = ss_observed.merge(ct_observed).merge(ah_observed)
+        observed_grid = self.gridcells.filterBounds(observed)
+
+        occupied_habitat = (
+            self.scl_image.select([HABITAT_AREA])
+            .clipToCollection(observed_grid)
+            .rename(OCCUPIED_HABITAT_AREA)
+        )
+        self.export_image_ee(occupied_habitat, "pothab/occupied_habitat")
+
+        area = ee.Image.pixelArea().divide(1000000).updateMask(self.watermask)
+        str_hab_area = area.updateMask(self.structural_habitat).rename(
+            STRUCTURAL_HABITAT_AREA
+        )
+        habitat_area = occupied_habitat.addBands(str_hab_area)
```

Comment:
How about set this to `self.habitat_area` (and initialize in `init`), and then remove the additional `habitat_area_image` arg in `attribute_areas`?